### PR TITLE
Re-add compilation from C (resolves #8)

### LINKED ISSUE #8: Files/directories not found in /tmp/pip-install-trzvk4fh/pybloomfiltermmap3/pip-egg-info
It's Failed when execute `pip install  pybloomfiltermmap3 inside docker`
```
Collecting ecdsa==0.13 (from -r /home/docker/requirements.txt (line 8))
  Downloading https://files.pythonhosted.org/packages/63/f4/73669d51825516ce8c43b816c0a6b64cd6eb71d08b99820c00792cb42222/ecdsa-0.13-py2.py3-none-any.whl (86kB)
Collecting Cython==0.29.13 (from -r /home/docker/requirements.txt (line 9))
  Downloading https://files.pythonhosted.org/packages/45/f2/a7101b3457561e57f5abcd6f5ac13190054fecd7370f58f36fe2d6574742/Cython-0.29.13-cp36-cp36m-manylinux1_x86_64.whl (2.1MB)
Collecting pybloomfiltermmap3==0.4.17 (from -r /home/docker/requirements.txt (line 10))
  Downloading https://files.pythonhosted.org/packages/b8/c1/8b1c7b0db349e9a8d39cdc978e843c54302d6678adf096a32daa2a20987f/pybloomfiltermmap3-0.4.17.tar.gz (472kB)
ERROR: Files/directories not found in /tmp/pip-install-trzvk4fh/pybloomfiltermmap3/pip-egg-info
WARNING: You are using pip version 19.1.1, however version 19.2.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

```

### PR DESCRIPTION
**Pull request by mizvyt**
This should finally solve the dependency problem with Cython (#8)
Also fixed some of the typos in README, added contribution instructions.

Tried this by deploying into TestPyPI, and then pulling the package in a clean virtual environment in both local setup and a Docker deployment.

Note: only issue I ran into was uploading a bdist_wheel (built in ubuntu 18.04.03 x64) to TestPyPI:
```
HTTPError: 400 Client Error: Binary wheel 'pybloomfiltermmap3-0.4.19-cp36-cp36m-linux_x86_64.whl' has an unsupported platform tag 'linux_x86_64'. for url: https://test.pypi.org/legacy/
```
This may have to do with the legacy pypi repo, but from what I saw it seems that it only supports multi-OS dists. Let me know if you have any idea about this.

### CODE REVIEW

**Comment by mizvyt** [comment]
Note: a test in previous commit failed (https://travis-ci.org/prashnts/pybloomfiltermmap3/jobs/602627965)
```
FAIL: test_accuracy (tests.accuracytest.StringAccuracyMallocTestCase)
AssertionError: False is not true : false_pos: 21 / 10000 = 0.0021 > 0.002
```
Any idea why this may fail? A bug somewhere in the code that deals with float values?

**Comment by willbowditch** [comment]
This PR fixes installation problems I was having on OSX

**Comment by mizvyt** [comment]
bump @prashnts

**Comment by prashnts** [comment]
It looks good, except the indentations which were changed to 4 spaces :( 

I hope it's fine if I revert the tab size to 2 spaces to be consistent @mizvyt? :) 

> FAIL: test_accuracy (tests.accuracytest.StringAccuracyMallocTestCase)
> AssertionError: False is not true : false_pos: 21 / 10000 = 0.0021 > 0.002

I've seen it few times. I suspect it's travis ci where syscalls can be throttled (since its running in a vm). Usually restarting the test works.

**Comment by prashnts** [comment]
Actually never mind, just realised that rest of the codebase uses 4 space tabs, so let's make `setup.py` consistent as well.

@mizvyt would you like to have maintainer access to the repo? I missed several of your (very useful) PRs due to lack of time.

File: README.markdown
Comment (prashnts) [inline README.markdown:38]: Now that we are sort of diverging the repo from old implementation, perhaps we should host the docs too?

File: setup.py
Comment (prashnts) [inline setup.py]: This can be very ambiguous, since `ImportError` can be either the one you raise in line 28, or from 30.

Don't use exceptions for case-matching. Instead branch out. A refactor block is below:

```py
# branch out based on `--no-cython` in `argv`.
# assume `--cython` as default to avoid havig to deal with both params being there.

if "--no-cython" in sys.argv:
	# note that we let the exception bubble up if `pybloomfilter.c` doesnt exist.
	ext_files.append("src/pybloomfilter.c")
else:
	from Cython.Distutils import build_ext

	ext_files.append("src/pybloomfilter.pyx")
	setup_kwargs["cmdclass"] = {"build_ext": build_ext}
```

**Review by prashnts — CHANGES_REQUESTED** [review]
`setup.py` needs some refactor.

File: README.markdown
Comment (mizvyt) [reply README.markdown:38]: Good idea.
I suggest taking over this domain to host:
https://pybloomfilter.readthedocs.io/en/latest/
It is my belief that any new users should be diving into the python 3 version.. :-)

I can see that this one has something published, is this yours?:
https://pybloomfiltermmap3.readthedocs.io/en/latest/

File: setup.py
Comment (mizvyt) [reply setup.py]: You're right, it makes sense - I have mostly left this code as is when I pushed the changes.

**Comment by mizvyt** [comment]
> @mizvyt would you like to have maintainer access to the repo? I missed several of your (very useful) PRs due to lack of time.

Sure, send me an email. I wouldn't mind :-)

File: setup.py
Comment (mizvyt) [reply setup.py]: ca3d656

File: README.markdown
Comment (prashnts) [reply README.markdown:38]: Ah! Yes, it's me. Its last build was 2 year 9 months ago, and I forgot about it. xD

I'll set it up again tomorrow.

**Review by prashnts — APPROVED** [review]
Looks good now, and the hint about adding the parameters in error message is a nice touch!

**Comment by prashnts** [comment]
I added you as a collaborator to this repo @mizvyt! I'll let you merge this PR as tiny "welcome" gesture! :)

**Pull request merged**
